packages/yew-router/src/hidden.rs

```diff
@@ -0,0 +1,407 @@
+use std::any::{Any, TypeId};
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::fmt::Display;
+use std::mem;
+use std::str::FromStr;
+
+use percent_encoding::{percent_decode_str, percent_encode, NON_ALPHANUMERIC};
+use regex::RegexSet;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use wasm_bindgen::JsValue;
+
+use crate::history::Route;
+pub use crate::Routable;
+
+/// This trait is implemented by the derive macro. There is a blanket implementation of
+/// `Routeable` for all `DerivedRoutable`.
+pub trait DerivedRoutable: PartialEq + Any + Default + Clone {
+    const VARIANTS: &'static [RouteMuncher<Self>];
+
+    fn to_args(&self) -> (usize, Args);
+}
+
+/// Lazily builds and caches a `RegexSet` keyed on the TypeId of the `DerivedRoutable`.
+fn cached_regex_set<T: DerivedRoutable>() -> &'static RegexSet {
```

Comment:
Have you tried benchmarking performance with and without regex? I think we should leave out regex parsing at runtime because of it's performance implications (poor performance first time is a major issue)

Reply:
I did some performance measurements whilst looking into #1885 and regex did not show up at all, but I'll do some more tests.

Reply:
On a second thought, how about we remove the runtime regex optimization for now and look at it later? I think we should roll these changes in then benchmark/optimize it.

Reply:
I can do that - I'll need to modify the `PathMuncher` a little to make sure it matches the end of the route if there's no `#[bind(rest)]` field, as currently it relies on the regex for this, but it's only a minor change.

Reply:
I've done the above. Instead of removing regex support completely, I've put it behind a disabled-by-default feature flag. This will make it easier to benchmark with/without the regex optimisation and potentially re-enable it by default in future.